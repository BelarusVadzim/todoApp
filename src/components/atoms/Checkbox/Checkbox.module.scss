@import 'src/variables.scss';

.checkboxBase {
  display: block !important;
  cursor: pointer;
  font-size: $font-size !important;
  user-select: none;
  border-radius: 2px;
  outline-color: $primary-color;
}

.checkboxBase:focus-within:not(:active) {
  outline-style: solid;
  outline-width: 2px;
}

.checkboxColored {
  background: linear-gradient($primary-color, #e058ec);
  background-clip: text;
  -webkit-text-fill-color: transparent;
}

.checkboxChecked {
  @extend .checkboxBase;
  @extend .checkboxColored;
}

.checkboxUnchecked {
  color: $inactive-color;

  @extend .checkboxBase;

  &:hover {
    @extend .checkboxColored;
  }
}

.input {
  height: 0;
  width: 0;
  opacity: 0;
}

