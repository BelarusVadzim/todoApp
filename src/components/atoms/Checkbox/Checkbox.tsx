import React from 'react';
import style from './Checkbox.module.scss';
import { PropsWithClassName } from 'types';

type CheckboxProps = PropsWithClassName & {
  checked?: boolean;
  onToggle?: () => void;
};

const Checkbox: React.FC<CheckboxProps> = ({
  checked = false,
  onToggle,
  className,
}) => {
  const materialIconName = checked ? 'check_circle' : 'radio_button_unchecked';
  let combinedClassName = checked
    ? style.checkboxChecked
    : style.checkboxUnchecked;
  combinedClassName = className
    ? `${combinedClassName} ${className}`
    : combinedClassName;

  return (
    <div className={`material-icons ${combinedClassName}`} onClick={onToggle}>
      {materialIconName}
    </div>
  );
};

export default Checkbox;
