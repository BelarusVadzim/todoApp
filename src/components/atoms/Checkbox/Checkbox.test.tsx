import { render } from '@testing-library/react';
import Checkbox from './Checkbox';

describe('<Checkbox />', () => {
  it('should render properly', () => {
    const component = render(<Checkbox />);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('when checked and classname specified should render properly ', () => {
    const component = render(<Checkbox checked className="extClassName" />);

    expect(component.asFragment()).toMatchSnapshot();
  });
});
