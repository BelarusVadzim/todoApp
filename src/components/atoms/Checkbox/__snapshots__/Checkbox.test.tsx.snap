// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="material-icons checkboxUnchecked"
  >
    radio_button_unchecked
    <input
      hidden=""
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;

exports[`<Checkbox /> when checked and classname specified should render properly  1`] = `
<DocumentFragment>
  <div
    class="material-icons checkboxChecked extClassName"
  >
    check_circle
    <input
      checked=""
      hidden=""
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;
